propagation.go

```diff
@@ -41,71 +45,115 @@ var (
 type BuiltinFormat byte
 
 const (
-	// SplitBinary encodes the Span in a SplitBinaryCarrier instance.
+	// Binary encodes the Span for propagation as opaque binary data.
+	//
+	// For Tracer.Inject(): the carrier must be an `io.Writer`.
 	//
-	// The `carrier` for injection and extraction must be a
-	// `*SplitBinaryCarrier` instance.
-	SplitBinary BuiltinFormat = iota
+	// For Tracer.Join(): the carrier must be an `io.Reader`.
+	Binary BuiltinFormat = iota
 
-	// SplitText encodes the Span in a SplitTextCarrier instance.
+	// TextMap encodes the Span in a TextMapCarrier instance.
+	//
+	// For Tracer.Inject(): the carrier must be a `TextMapWriter`.
+	//
+	// For Tracer.Join(): the carrier must be a `TextMapReader`.
+	//
+	// See HTTPHeaderTextMapCarrier for an implementation of both TextMapWriter
+	// and TextMapReader that defers to an http.Header instance for storage.
+	// For example, Inject():
+	//
+	//    carrier := HTTPHeaderTextMapCarrier(httpReq.Header)
+	//    err := span.Tracer().Inject(span, TextMap, carrier)
 	//
-	// The `carrier` for injection and extraction must be a `*SplitTextCarrier`
-	// instance.
+	// Or Join():
+	//
+	//    carrier := HTTPHeaderTextMapCarrier(httpReq.Header)
+	//    span, err := tracer.Join("opName", TextMap, carrier)
+	//
+	TextMap
+
+	// SplitBinary is DEPRECATED
+	SplitBinary
+	// SplitText is DEPRECATED
 	SplitText
+)
+
+// TextMapWriter is the Inject() carrier for the TextMap builtin format. With
+// it, the caller can encode a Span for propagation as entries in a multimap of
+// unicode strings.
+type TextMapWriter interface {
+	// Add a key:value pair to the carrier. Multiple values may be added for a
+	// single (repeated) key.
```

Comment:
@yurishkuro good point re the disclaimer... I'll add something. (This got lost during the revision where the carrier was handling the prefixing)